server-spi/src/main/java/org/keycloak/models/RealmProvider.java

```diff
@@ -81,16 +83,42 @@ default GroupModel createGroup(RealmModel realm, String name, GroupModel toParen
 
     RoleModel getRealmRole(RealmModel realm, String name);
 
-    Set<RoleModel> getRealmRoles(RealmModel realm);
-    
-    Set<RoleModel> getRealmRoles(RealmModel realm, Integer first, Integer max);
-    
-    Set<RoleModel> getClientRoles(RealmModel realm, ClientModel client, Integer first, Integer max);
-    
-    Set<RoleModel> searchForClientRoles(RealmModel realm, ClientModel client, String search, Integer first,
-            Integer max);
-    
-    Set<RoleModel> searchForRoles(RealmModel realm, String search, Integer first, Integer max);
+    @Deprecated
+    default Set<RoleModel> getRealmRoles(RealmModel realm) {
+        return getRealmRolesStream(realm).collect(Collectors.toSet());
+    }
+
+    Stream<RoleModel> getRealmRolesStream(RealmModel realm);
+
+    @Deprecated
+    default Set<RoleModel> getRealmRoles(RealmModel realm, Integer first, Integer max) {
+        return getRealmRolesStream(realm, first, max).collect(Collectors.toSet());
+    }
+
+    Stream<RoleModel> getRealmRolesStream(RealmModel realm, Integer first, Integer max);
```

Comment:
Should also be methods for accessing roles within scope of this work?

Reply:
It was simpler to include also `getRealmRolesStream` because there is a common method `getRolesStream` in `JpaRealmProvider` which operates with both `getClientRoles` and `getRealmRoles`.  So when I changed return type of `getRolesStream` to Stream<> it wasn't possible to call this method from `getClientRoles` and `getRealmRoles` if they both don't operate with streams.